README.md

```diff
@@ -632,37 +632,42 @@ enum AgentCapabilities {
 
 <h3 id="agentdescription-message">AgentDescription Message</h3>
 
-
 The AgentDescription message has the following structure:
 
-
 ```protobuf
 message AgentDescription {
-    string agent_type = 1;
-    string agent_version = 3;
-    repeated KeyValue agent_attributes = 4;
+    repeated KeyValue identifying_attributes = 1;
+    repeated KeyValue non_identifying_attributes = 2;
 }
 ```
 
+#### identifying_attributes
 
-<h4 id="agent_type">agent_type</h4>
-
-
-The reverse FQDN that uniquely identifies the agent type, e.g.
-"io.opentelemetry.collector".
+Attributes that identify the agent. Keys/values are according to OpenTelemetry
+semantic conventions, see:
+https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions
 
-<h4 id="agent_version">agent_version</h4>
+For standalone running Agents (such as OpenTelemetry Collector) the following
+attributes SHOULD be specified:
 
+- service.name should be set to a reverse FQDN that uniquely identifies the
```

Comment:
Let's say that the agent in question is OpenTelemetry Java Instrumentation. How can we inform the server about this? 

[service.name](https://github.com/open-telemetry/opentelemetry-specification/blob/da74730bf835010229a9612c281f052e26553218/specification/resource/semantic_conventions/README.md#service) could be already set to something like `shoppingcart` and I am thinking that using it for `io.opentelemetry.collector` might not be a good fit here. 

Perhaps `service.name` could be optional and we should have a new convention (`service.type`?). It could hold values such as `opentelemetry-collector`, `fluentbit`, `opentelemetry/java-auto-instrumentation` or such

Reply:
Java Instrumentation will not use `service.name` to describe itself. `service.*` attributes are intended to describe the Application (or the Service) that is being instrumented, not the SDK or the library that does the instrumentation.

Java instrumentation is part of an Otel SDK and Otel SDK uses a different set of attributes to describe itself. We already have some [conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#telemetry-sdk) for the SDKs. Java Instrumentation will set:
telemetry.sdk.name="opentelemetry"
telemetry.sdk.language="java"
telemetry.sdk.version="some version number" (and/or `telemetry.auto.version`).

These are probably not sufficient though. We will likely also want to have a unique identifier that identifies the particular running instance of the application instrumented with Java SDK. This is important if we want SDKs or instrumentation libraries to report their own telemetry (e.g. send metrics about the number of telemetry items they collected). For this I think we can add a convention `telemetry.sdk.instance.id` (by analogy of `service.instance.id`). SDKs will autogenerate the instance id at startup or will allow to specify it via an env variable if persistent id is wanted. Or perhaps the SDK could also set `telemetry.sdk.instance.id` equal to `service.instance.id` if the latter is provided but the former is not provided.

Reply:
My point is that `service.name` here is said to "uniquely identify the agent type, e.g. 'io.opentelemetry.collector'" which I think is not true. `service.name` has different purpose and we might want to have some other field to identify the type. 

Perhaps this could be done via checking fields like `telemetry.sdk.*` (for SDK) but still it does not say how to detect that the agent is, say, OpenTelemetry Collector or Fluentbit. We had `agent_type` field previously, which is now gone

Reply:
So `service.name=io.opentelemetry.collector` for OpenTelemetry Collector, `service.name=io.fluentbit` for Fluentbit and so forth. If the value is unknown (e.g. `service.name=foo`), it indicates it's an SDK and one needs to look to `telemetry.sdk.*` to understand what kind of SDK is being employed?

Reply:
>So service.name=io.opentelemetry.collector for OpenTelemetry Collector, service.name=io.fluentbit for Fluentbit and so forth.

Yes.

>If the value is unknown (e.g. service.name=foo), it indicates it's an SDK and one needs to look to telemetry.sdk.* to understand what kind of SDK is being employed?

No. If `service.name` but the value is unknown then it is still valid and the server should assume it is a standalone Agent, just the kind of an Agent that is unknown to the Server (what Server does in that case is Server's business and is outside of the protocol spec).

The Server generally doesn't need to look for any specific attribute (`service.*` or `telemetry.sdk.*`). It should just:
1. Use what is listed in `identifying_attributes` as a set of attributes to link to the Agent's own telemetry sent elsewhere.
2. Use a `identifying_attributes` and `non_identifying_attributes` as a set of attributes over which the Server may define any business logic / rules to return appropriate configuration to the Agent.

The Server doesn't necessarily need to have any hard-coded logic for specific values of `service.name`, although it can have such logic if wanted. The Server may for example have a specific hard-coded logic for the case service.name="io.opentelemetry.collector" and another hard-coded for for the case when telemetry.sdk.name="opentelemetry". Again, these are implementation details of the Server, they are not protocol concerns.

Real-world Server implementations will likely employ a combination of hard-coded rules (for specific attributes values), generic rules (e.g. for linking to own telemetry of arbitrary agents without know what agent it is) and user-defined rules (e.g. to allow users to override config for specific agents or subsets of agents).